src/accounts/accounts.service.ts

```diff
@@ -206,4 +206,39 @@ export class AccountsService {
     await this.accountRepository.update({ id: update.id }, currentUser);
     return [currentUser, null];
   }
+
+  async removeAccount(id: string, curAccount: Account) {
+    if (curAccount.id === id) return [null, "You can't remove self account"];
+
+    const removeAccount = await this.accountRepository.findOne({
+      where: {
+        id: id,
+      },
+      select: ['id', 'role', 'isActive'],
+    });
+
+    if (!removeAccount) {
+      return [null, "Can't find account to remove"];
+    }
+    if (removeAccount.role === RoleEnum.ADMIN)
+      return [null, "You can't remove admin account"];
+    if (
+      curAccount.role === RoleEnum.USER ||
+      (curAccount.role === RoleEnum.MANAGER &&
+        removeAccount.role !== RoleEnum.USER)
+    ) {
+      return [null, 'Account role must be higher to remove this account'];
+    }
+    if (removeAccount.isActive) {
```

Comment:
This is not right requitement, when account is deactive then remove, else don't

Reply:
else return this account already active

Reply:
Check account is active first, if active return error, then check role, it will more faster in process